src/globals.c

```diff
@@ -727,7 +784,7 @@ void global_init()
     PH1_IR = malloc(MAX_IR_INSTR * sizeof(ph1_ir_t));
     PH2_IR = malloc(MAX_IR_INSTR * sizeof(ph2_ir_t));
     LABEL_LUT = malloc(MAX_LABEL * sizeof(label_lut_t));
-    SOURCE = malloc(MAX_SOURCE);
+    SOURCE = create_source(MAX_SOURCE);
```

Comment:
I'm thinking of choosing an appropriate power of 2 for `MAX_SOURCE`. perhaps `2^18 (262144 B, or 256 KiB)` or `2^19 (524288 B, or 512 KiB)` would be better?

Reply:
Then, we do need scientific ways to calculate.